assembl/static2/js/app/components/login/sendPwdForm.jsx

```diff
@@ -2,27 +2,64 @@ import React from 'react';
 import { connect } from 'react-redux';
 import { Translate, I18n } from 'react-redux-i18n';
 import { form, FormGroup, FormControl, Button } from 'react-bootstrap';
+import inputHandler from '../../utils/inputHandler';
+import { requestPasswordChangeAction } from '../../actions/authenticationActions';
 
 class SendPwdForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { identifier: null };
+    this.submitHandler = this.submitHandler.bind(this);
+    this.handleInput = this.handleInput.bind(this);
+    this.messageHandler = this.messageHandler.bind(this);
+  }
+
+  handleInput(e) {
+    inputHandler(this, e);
+    const el = document.getElementById('sendPwdForm-error');
+    if (el) { el.innerHTML = ''; }  // BUG: Only shows once; if fails beyond first time, no error shown!
+  }
+
+  messageHandler() {
+    return (
+      <div>
+        {
+          this.props.auth.passwordChangeRequest.success == null ? <span /> :
+          <div id="sendPwdForm-error"><Translate value="login.passwordChangeRequestError" /></div>
+        }
+      </div>
+    );
+  }
+
+  submitHandler(e) {
+    e.preventDefault();
+    this.props.sendRequest(this.state.identifier);
+  }
+
   render() {
-    const { debateData } = this.props.debate;
-    const { rootPath } = this.props.context;
     return (
       <div className="login-view">
         <div className="box-title">
           <Translate value="login.forgotPwd" />
         </div>
         <div className="box">
-          <form className="resendPwd" method="POST" action={`${rootPath}${debateData.slug}/changePassword#success`}>
+          <form className="resendPwd" onSubmit={this.submitHandler}>
+            <input type="hidden" name="referer" value="v2" />
```

Comment:
This was the initial implementation that Quentin started. I've kept it for compatibility, but the backend (I don't think) makes further checks for this. I'll double check, if it doesn't, I'll remove this, otherwise, it is required to have in order to land on the /home page after a login.